### PR TITLE
Allow to access the fiber local storage outside of the monad

### PR DESCRIPTION
**Pull request by diml**
This will allow to memoize functions that do not return a Fiber.t, which will make it much easier to use the new memoization system.

### CODE REVIEW

File: src/lib.ml [hunk 1]
```diff
@@ -1179,6 +1179,13 @@ let report_lib_error ppf (e : Error.t) =
       Lib_name.pp_quoted not_vlib.name
       Lib_name.pp_quoted impl.name
 
+let () =
```
Comment (rgrinberg) [inline src/lib.ml]: Why do we need to report the error here? Don't we already do it in report_errors?

File: src/memo/memo.ml [hunk 2]
```diff
@@ -204,25 +204,19 @@ let global_dep_dag = Dag.create ()
...
-                 (Stack_frame.name st)
-                 (Stack_frame.input st |> Sexp.to_string)
-   )
-  )
-  >>| (fun _ -> v)
+let dump_stack () =
+  let stack = get_call_stack () in
+  Printf.printf "Memoized function stack:\n";
+  List.iter stack ~f:(
+    fun st -> Printf.printf "   %s %s\n"
```
Comment (rgrinberg) [inline src/memo/memo.ml]: I just noticed this, but we should probably print this to stderr. This is how we log everything as well.

Comment (diml) [reply src/memo/memo.ml]: Good call

Comment (diml) [reply src/memo/memo.ml]: Fixed

**Pull request merged**